src/server/object_services/object_server.js

```diff
@@ -1415,9 +1417,35 @@ async function find_object_md(req) {
         obj = await MDStore.instance().find_object_latest(req.bucket._id, req.rpc_params.key);
     }
     check_object_mode(req, obj, 'NO_SUCH_OBJECT');
+    await calculate_multipart_range(req.rpc_params.part_number, obj);
+
     return obj;
 }
 
+/**
+ * calculate (start, end) range of the specified part_number
+ *
+ * @param part_number integer in 1..10000 range
+ * @param {nb.ObjectMD} obj
+ */
+async function calculate_multipart_range(part_number, obj) {
+    if (!part_number) {
+        return; // part number was not specified in the S3 Query
+    }
+
+    // fetch the multiparts of this obj which are lte(<=) the requested part number
+    const multiparts = await MDStore.instance().find_lte_multiparts_of_object_by_number(obj._id, part_number);
```

Comment:
Can you at least add a comment with todo:
```javascript
// TODO store the range inside each multipart in _complete_object_multiparts() so we can read just the requested multipart here
```

Reply:
Sure